bundles/org.openhab.binding.kostalinverter/src/main/java/org/openhab/binding/kostalinverter/internal/secondgeneration/SecondGenerationHandler.java

```diff
@@ -172,17 +211,12 @@ public void initialize() {
                 updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
                         scheduleWithFixedDelayException.getClass().getName() + ":"
                                 + scheduleWithFixedDelayException.getMessage());
-            } catch (InterruptedException interruptedException) {
-                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
-                        interruptedException.getClass().getName() + ":" + interruptedException.getMessage());
-            } catch (ExecutionException executionException) {
-                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
-                        executionException.getClass().getName() + ":" + executionException.getMessage());
-            } catch (TimeoutException timeoutException) {
+            } catch (Exception handleCommandException) {
```

Comment:
Please do not catch Exception but rather a list of more specific exceptions like for example: `InterruptedException  | ExecutionException  | TimeoutException e`